src/components/FilterList.tsx

```diff
@@ -0,0 +1,30 @@
+import { ChangeEvent } from "react";
+
+interface FilterListProps {
+	searchTerm: string;
+	setSearchTerm: (term: string) => void;
+}
+
+export function FilterList({ searchTerm, setSearchTerm }: FilterListProps) {
+	const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+		setSearchTerm(e.target.value);
```

Comment:
I noticed when I console.log(searchTerm) after the state update, the searchTerm is missing a letter. I think using a useEffect to obtain the latest updated state after the user inputs a search term would be helpful  to get around React's asynchronous nature of updating states and always maintain the latest state value immediately!

useEffect(() => {
        console.log('Updated searchTerm:', searchTerm);
    }, [searchTerm]);

Reply:
I'm having a hard time reproducing this on my end. React _should_ be handling this for us already 👀

Reply:
Uploading Screen Recording 2024-09-01 at 12.30.47 AM.mov…

Reply:
I dont think its really an issue, but its more an observation

https://github.com/user-attachments/assets/c15e31e6-883a-44e0-be00-d2ccedef4a27

Reply:
I'm also not sure I can reproduce this on my end 🤔

Reply:
Thanks for the screen record! I did get too when I have been trying to recreate it. I didn't think it was behind in state, which maybe is on my part. I thought it was working as react intended, because the `console.log(searchTerm)` will log the state before the the update takes effect, because the function has to complete before they match, but to my understanding it does filter based on the full input.

Reply:
Ahhh gotcha!! That makes sense then.  Natural React behavior. And yes! It is doing the thing its supposed to do exactly: filter the list every single time!!! Great job on the refactoring changes by the way.  I thought I'd bring it up in case I was missing something..